wrapper.go

```diff
@@ -53,3 +54,31 @@ type KeyExporter interface {
 	// KeyBytes returns the "current" key bytes
 	KeyBytes(context.Context) ([]byte, error)
 }
+
+// CryptoProvider provides access to a set of keys within a certain pool,
+// e.g. a PKCS#11 token slot. Specific keys with varying capabilities can
+// be accessed using GetKey.
+type CryptoProvider interface {
```

Comment:
Separately, I wonder if we can make `keysutil.LockManager` implement `CryptoProvider`... I suppose `GetKey(...)` will need versions and some other information, but it might be doable outside of derived keys?